### PR TITLE
GIVCAMP-23 | Storyblok integration

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW

# Summary
- Initial setup with Storyblok (with their starter test components)
- Configure the environment variable plugin
- Readme

# Review By (Date)
- Soon

# Review Tasks

## Setup tasks and/or behavior to test

1. Go to https://deploy-preview-7--giving-campaign.netlify.app/ (I apologize for the extremely janky setup 😂 )
https://deploy-preview-7--giving-campaign.netlify.app/about-test
https://deploy-preview-7--giving-campaign.netlify.app/randomthingsmonkey (click to preview the 404 page)
2. Look at code and things installed - this is just the initial bare minimal batch. Have tickets to install Decanter/TW and other supporting packages after this one. Will also look into Slices 🍕 

# Associated Issues and/or People
- JIRA ticket(s) - GIVCAMP-23

### CODE REVIEW

**Comment by yvonnetangsu** [comment]
@sherakama
I expected the built to fail since the new Storyblok tokens hasn't been added to the vault yet, but the log's telling me that `Error: Failed to authenticate to vault`. I think I set up the netlify.toml file like on SAA homesite, but am I missing anything? 🤔 I thought the vault secret and role id are added on Netlify at the top level so it should be able to pick them out automatically (could be totally wrong)?

Screen Shot 2023-02-22 at 11 40 20 AM

**Comment by sherakama** [comment]
Screenshot 2023-02-22 at 11 48 09

I believe you have to `opt-in` to the new style. Let me try that for you.

**Comment by yvonnetangsu** [comment]
Oh interesting 👀  Thank you @sherakama! Hope the training is fun 😄

**Comment by sherakama** [comment]
Looks like you're authenticated with vault now but have new env var build problems. Good luck.

**Comment by yvonnetangsu** [comment]
> Looks like you're authenticated with vault now but have new env var build problems. Good luck.

Yeah, Rebecca is working on that ticket to replace the variables on vault with the new ones so that is within expectation. Thank you @sherakama for your help 😄 ⭐ !!

**Comment by rebeccahongsf** [comment]
@yvonnetangsu it should be working now! 😄  https://deploy-preview-7--giving-campaign.netlify.app/

**Comment by yvonnetangsu** [comment]
Awesome thank you @rebeccahongsf vault manager 😄  🚀 ⭐

**Comment by sherakama** [comment]
Just have a quick question. Is there any reason why @yvonnetangsu is not also changing vault vars?

**Comment by yvonnetangsu** [comment]
> Just have a quick question. Is there any reason why @yvonnetangsu is not also changing vault vars?

Oh we tried to divide up the initial batch of tickets during the initial (sort of) sprint planning, and @rebeccahongsf took on this one - no particular reason 😄

File: netlify.toml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# BUILD Settings:
+# ##############################################################################
+[build]
+  publish = "public"
+
+[dev]
+  publish = "public"
+  port = 64946
+  autolaunch = true
```
Comment (yvonnetangsu) [inline netlify.toml:9]: I can remove this if we don't want to support `netlify dev` I think we stopped using it for Giving and SAA site?

Comment (sherakama) [reply netlify.toml:9]: It comes in handy every once in a while when trying to troubleshoot something in a build or function but we very rarely need to use it now.

Comment (yvonnetangsu) [reply netlify.toml:9]: I see - I'll probably remove it in the next PR then. We can always add it back if we need it later. Trying to be good about only adding things that we use 😄

Comment (yvonnetangsu) [reply netlify.toml:9]: PR ok to merge?

**Review by rebeccahongsf — APPROVED** [review]
Netlify preview built successfully and displays in the [SB authoring env](https://app.storyblok.com/#/me/spaces/1005200/stories/0/0/38938). I'd defer to @sherakama for guidance on the `netlify.toml` but otherwise, this look merge ready to me! 🖖

**Comment by yvonnetangsu** [comment]
Thanks for the approval @rebeccahongsf. @sherakama I'll continue to add/change things so please leave any comments here!

Moving over to MyAccount/Directory for a while - got some stuff to take care of 🖖🏼

**Comment by yvonnetangsu** [comment]
Oooo merging with the merge queue "Confirm merge when ready"

File: package.json [hunk 2]
```diff
@@ -19,12 +18,14 @@
     "not dead"
   ],
   "scripts": {
-    "develop": "gatsby develop",
-    "dev": "gatsby develop",
-    "start": "gatsby develop",
+    "develop": "ENABLE_GATSBY_REFRESH_ENDPOINT=true gatsby develop",
+    "dev": "ENABLE_GATSBY_REFRESH_ENDPOINT=true gatsby develop",
+    "start": "ENABLE_GATSBY_REFRESH_ENDPOINT=true gatsby develop",
```
Comment (sherakama) [inline package.json:23]: This is three different ways to do the same thing. Can we just have `dev`?

File: src/components/layout.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import Feature from "./Feature";
+
+type LayoutProps = {
+  children: React.ReactNode,
+}
+
+storyblokInit({
+  accessToken: process.env.STORYBLOK_ACCESS_TOKEN,
+  apiOptions: {
+    region: "us", // Pass this key/value if your space was created under US region
```
Comment (sherakama) [inline src/components/layout.tsx:14]: That's it? Seems so easy.

File: package.json (hunk 2 above)
Comment (yvonnetangsu) [reply package.json:23]: Sure 😂  Was thinking maybe any new devs come to the project is used to typing it one way or another, but totally not needed

File: src/components/layout.tsx (hunk 3 above)
Comment (yvonnetangsu) [reply src/components/layout.tsx:14]: Seems ok so far 🤷🏼‍♀️

**Pull request merged**